src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -13,38 +13,170 @@
 using Akka.Configuration;
 using Akka.Hosting;
 
+#nullable enable
 namespace Akka.Cluster.Hosting
 {
+    public sealed class Role
+    {
+        public Role(string name, int? minimumNumberOfMembers = null)
+        {
+            Name = name;
+            MinimumNumberOfMembers = minimumNumberOfMembers;
+        }
+
+        /// <summary>
+        ///     The role name
+        /// </summary>
+        public string Name { get; }
+        
+        /// <summary>
+        ///     <para>
+        ///         Minimum required number of members before the leader changes member status
+        ///         of 'Joining' members to 'Up'. Typically used together with
+        ///         <see cref="Cluster.RegisterOnMemberUp"/> to defer some action, such as starting actors,
+        ///         until the cluster has reached a certain size.
+        ///     </para>
+        ///     <b>Default:</b> 1
+        /// </summary>
+        public int? MinimumNumberOfMembers { get; set; }
+    }
+    
     /// <summary>
-    /// The set of options for enabling Akka.Cluster support.
+    ///     The set of options for enabling Akka.Cluster support.
     /// </summary>
     public sealed class ClusterOptions
     {
         /// <summary>
-        /// The akka.cluster.roles values.
+        ///     The akka.cluster.roles and akka.cluster.role values.
         /// </summary>
-        public string[] Roles { get; set; }
+        public Role[]? Roles { get; set; }
 
         /// <summary>
-        /// If populated, the akka.cluster.seed-nodes that will be used.
+        ///     If populated, the akka.cluster.seed-nodes that will be used.
+        /// </summary>
+        public Address[]? SeedNodes { get; set; }
```

Comment:
So here's a goal worth considering with the `ClusterOptions` class - should this be parseable from an `appsettings.json` file / Environment Variable? If so, maybe we should support `string` here instead and just handle the parsing on the back-end. I'm leaning towards doing that for 1.0.

Reply:
Yeah, let's change this to a `string` type

Reply:
Here's what we have to do in the current stable version of Akka.Hosting:

```csharp
var akkaSection = context.Configuration.GetSection("Akka");

// maps to environment variable Akka__ClusterIp
var hostName = akkaSection.GetValue<string>("ClusterIp", "localhost");

// maps to environment variable Akka__ClusterPort
var port = akkaSection.GetValue<int>("ClusterPort", 0);

var seeds = akkaSection.GetValue<string[]>("ClusterSeeds", new[] { "akka.tcp://SqlSharding@localhost:7918" })
    .Select(Address.Parse)
    .ToArray();
```

It would be amazing if we could just do this instead (and if we decide to go down this route, we should write a unit test that looks like this):

**`appsettings.json`**

```json
{
    "Logging": {
        "LogLevel": {
            "Default": "Debug",
            "System": "Information",
            "Microsoft": "Information"
        }
    },
    "Akka": {
        "RemoteOptions":{
            "Hostname": "localhost",
            "Port": 8081,
            "PublicHostname": "localhost",
            "PublicPort": 8081
        },
        "ClusterOptions": {
            "SeedNodes": "akka.tcp://ClusterSystem@localhost:8081",
            "Roles": ["backend"]
        }
    }
}
```

**`program.cs`**

```csharp
var akkaSection = context.Configuration.GetSection("Akka");
var clusterConfig = akkaSection.Get<ClusterOptions>(); // ideally, this should work
```

We've not really stated this as an explicit goal for the project before, but in the context of our "pit of success" mission for this library I think it's probably worth doing before we ship 1.0. What do you think?